pkg/gluster-virtblock/nodeserver.go

```diff
@@ -0,0 +1,159 @@
+package glustervirtblock
+
+import (
+	"context"
+	"os"
+	"path"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/gluster/gluster-csi-driver/pkg/utils"
+	"github.com/golang/glog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+// NodeServer struct of GlusterFS Block CSI driver with supported methods of CSI node server spec.
+type NodeServer struct {
+	*GfDriver
+}
+
+var mounter = mount.New("")
+
+// NodeStageVolume mounts the volume to a staging path on the node.
+func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
+// NodeUnstageVolume unstages the volume from the staging path
+func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "")
+}
+
+// NodePublishVolume mounts the volume mounted to the staging path to the target path
+func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	glog.V(2).Infof("received node publish volume request %+v", req)
+
+	err := ns.validateNodePublishVolumeReq(req)
+	if err != nil {
+		return nil, err
+	}
+
+	gs := req.GetVolumeContext()["glusterserver"]
+	volume := req.GetVolumeContext()["glustervol"]
+	hostPath, found := ns.Config.Mounts[volume]
+	if !found {
+		source := gs + ":" + volume
+		hostPath = path.Join(ns.Config.MntPathPrefix, volume)
+		err = utils.MountVolume(source, hostPath, "glusterfs", nil)
```

Comment:
What are the permissions of this volume ? how do we make sure the acl is properly configured between different app pod mounts ?

Reply:
Right now, it's mounted with default options. And with it host volume mount is accessible to different pods as per testing.